platform/platform_linux.cpp

```diff
@@ -171,20 +167,37 @@ std::string Platform::DeviceModel() const
 
 Platform::EConnectionType Platform::ConnectionStatus()
 {
-  int socketFd = socket(AF_INET, SOCK_STREAM, 0);
+  int socketFd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
   SCOPE_GUARD(closeSocket, std::bind(&close, socketFd));
   if (socketFd < 0)
     return EConnectionType::CONNECTION_NONE;
 
-  struct sockaddr_in addr;
-  memset(&addr, 0, sizeof(addr));
-  addr.sin_family = AF_INET;
-  addr.sin_port = htons(80);
-  inet_pton(AF_INET, kSomeWorkingWebServer, &addr.sin_addr);
+  struct addrinfo hints, *result, *p;
+  memset(&hints, 0, sizeof(struct addrinfo));
+  hints.ai_family = AF_INET;
+  hints.ai_socktype = SOCK_STREAM;
+  hints.ai_protocol = IPPROTO_TCP;
+  int res = getaddrinfo(METASERVER_HOST, METASERVER_SCHEME, &hints, &result);
```

Comment:
I don't like this approach. Resolving hostname can be blocked and it will hang the application. Let's stick back to the IP check instead.

Reply:
> Resolving hostname can be blocked and it will hang the application.

Fair point.
Only in this context, I thought the current codes' `connect` call is blocking already until the TCP handshake is over, which can take arbitrarily long depending on a number of factors, like what `TCP_TIMEOUT_INIT` was set to in kernel compilaiton time for example.
I think if we are worried about blocking here, then the current code is similarly worriesome.